examples/whisper/training/train.py

```diff
@@ -0,0 +1,165 @@
+    parser.add_argument("--learning_rate", type=float, default=1e-5, help="Learning rate for training.")
+    parser.add_argument("--gradient_accumulation_steps", type=int, default=1, help="Number of gradient accumulation steps.")
+    parser.add_argument("--fp16", action="store_true", help="Use mixed precision training.")
+    parser.add_argument("--eval_strategy", type=str, default="steps", help="Evaluation strategy to use during training.")
+    parser.add_argument("--save_steps", type=int, default=1000, help="Number of steps between model saves.")
+    parser.add_argument("--eval_steps", type=int, default=1000, help="Number of steps between evaluations.")
+    parser.add_argument("--logging_steps", type=int, default=25, help="Number of steps between logging.")
+    parser.add_argument("--push_to_hub", action="store_true", help="Whether to push the model to the Hugging Face Hub.")    
+    parser.add_argument("--per_device_train_batch_size", type=int, default=16, help="Batch size per device for training.")
+    parser.add_argument("--warmup_steps", type=int, default=500, help="Number of warmup steps for learning rate scheduler.")
+    parser.add_argument("--max_steps", type=int, default=5000, help="Maximum number of training steps.")
+    parser.add_argument("--dataloader_pin_memory", action="store_true", help="Whether to pin memory in the dataloader.")
+    parser.add_argument("--per_device_eval_batch_size", type=int, default=16, help="Batch size per device for evaluation.")
+    parser.add_argument("--generation_max_length", type=int, default=225, help="Maximum length for generation.")
+    parser.add_argument("--hub_model_id", type=str, default="baseten-admin/whisper_test1_hi", help="Hugging Face Hub model ID to push the trained model.")
+    parser.add_argument("--report_to", type=str, default="wandb", help="Reporting tool to use (e.g., 'wandb', 'tensorboard').")
+    parser.add_argument("--hub_strategy", type=str, default="end", help="Strategy for saving to the Hugging Face Hub.")
+
+    return parser.parse_args()
+
+def main(args):
+
+    # load model
+    model = WhisperForConditionalGeneration.from_pretrained(args.model_name)
+    model.generation_config.language = args.language
+    model.generation_config.task = args.task
+    model.generation_config.forced_decoder_ids = None
+
+    # To freeze decoder - TODO: verify 
+    # for param in model.decoder.parameters():
+    #     param.requires_grad = False
+
+    metric = evaluate.load("wer")
+    processor = WhisperProcessor.from_pretrained(args.model_name, language=args.language, task=args.task)
+
+    common_voice = load_and_process_dataset(args)
+
+    data_collator = DataCollatorSpeechSeq2SeqWithPadding(
+        processor=processor,
+        decoder_start_token_id=model.config.decoder_start_token_id,
+    )
+
+    training_args = Seq2SeqTrainingArguments(
+        output_dir=args.output_dir,  # change to a repo name of your choice
```

Comment:
we should default this to checkpoint dir and make sure that's a good default experience.